### DISCUSSION TITLE
Filter in-between dates

### DISCUSSION THREAD

**Discussion by mrbu-solutions** [pocketbase/pocketbase#2833] [Q&A] [answered]
Hi, thank you for this great framework, it makes development easier.

I have a scenario wherein I'm trying to display employees that are currently on leave today.

Here's the collection:

![image](https://github.com/pocketbase/pocketbase/assets/53817867/9b1023f9-5375-4973-b7a6-bced8927817d)

In the image above, the hard-coded date in the filter is today's date. They are both status approved.

I'm trying to filter 

1. status = approved; AND
2. today's date >= leaveStart; AND
3. today's date <= leaveEnd

I'm expecting the result should display the two highlighted record in the image but this is what I've got:
![image](https://github.com/pocketbase/pocketbase/assets/53817867/0464ed68-825a-4ffe-93c5-aaccc3a2f05c)

Is this a bug, or is there a better way to filter records that is in between 2 dates?

**Comment by mrbu-solutions** [comment]
Temporary solution:
`"('2023-06-28' > leaveStart || '2023-06-28' = leaveStart) && ('2023-06-28' < leaveEnd  || '2023-06-28' < leaveEnd) && status='approved'"`

![image](https://github.com/pocketbase/pocketbase/assets/53817867/fb8729ff-e2f0-497f-afd1-7f87a62cdd79)


It seems like greater or equal (>=) and less than or equal (<=) is not working

**Comment by ganigeorgiev** [comment, marked as answer]
The stored `date` field value is in the format `Y-m-d H:i:s.uZ`, meaning that if you want a range day filter, you'll have to specify the full date string format:

```js
status = "approved" && "2023-06-28 00:00:00.000Z" >= leaveStart && "2023-06-28 23:59:59.999Z" <= leaveEnd
```

_(use `00:00:00.000Z` or `23:59:59.999Z` to target the start/end of a day respectively)_

In the future there will be support for other date formats and normalizations like auto zero-ing partial datetime strings as in your case.

**Reply by mrbu-solutions** [reply]
Thank you very much!
